### DISCUSSION TITLE
Missing part in the results

### DISCUSSION THREAD

**Discussion by li2qn** [OpenRadioss/OpenRadioss#1566] [Users]
Hi everyone,
[New folder.zip](https://github.com/OpenRadioss/OpenRadioss/files/11724080/New.folder.zip)

I am new to Radioss and trying to do a drop test in Radioss. Currently my jobs run to completion, however in my results I only can see the rigid ground (using Rwall )but not the part (shell) that need to be dropped from certain distance and initial velocity. I am using ParaView for postprocessing. I am not sure why this is happening and struggling for a long time. Need help on this. Thanks so much.

Best.
Lee

**Comment by MarianBulla** [comment]
Hi li2qn,
can you double check the nodal coordinates of the node #4 and node #54, please ?
Once I modified them (to be on the plate...see attached file) and do some other small changes, your model is running and producing feasible results:
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/bf153b2e-def5-420b-8406-e0c3ba2be799)


Hope it helps you ?
[mod.zip](https://github.com/OpenRadioss/OpenRadioss/files/11724754/mod.zip)


Thanks and best
Marian

**Reply by li2qn** [reply]
Thank for your help Marian. May I ask which postprocessing software are you using? I tried  to run your modified file exactly, and post process it in ParaView.  Looks like only the Plate and Rigidwall shows in my results. However the scale of the rigidwall looks look infinite compare to the shell plate. is there anyway in the keywords line to limited the size or scale the size of the rigidwall?

![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/383a9105-5f64-40c5-b32a-d55034f2c3ea)

and zoom in view
![image](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/20ba0b6f-4c9f-4cb2-a9e0-2a810e729ce9)




Regrads
Lee

**Comment by MarianBulla** [comment]
Hi Lee,
I'm using HyperView as postprocessort.
Anyway, there are several ways to improve your postprocessing.
The easiest one is not to use the infinite RWall but switching to a limit sized RWall (parallelogram):
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/d585bf6b-c130-46da-b7c0-37b795a027b2)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/18b98545-1f05-4e00-ad4f-18b210319ad2)


[mod2.zip](https://github.com/OpenRadioss/OpenRadioss/files/11731632/mod2.zip)

Hope it helps you ?

Thanks and best regards,
Marian

**Comment by li2qn** [comment]
Many Thanks Marian. That's very helpful. May I ask does Radioss have a keyword that perform general contact like self contact in ABAQUS? In that way I don't need to specify each contact pairs. Thanks again for your time and discussion.

Regrads
Lee

**Reply by PaulAltair** [reply]
The contact interfaces in OpenRadioss are very flexible, although they are described as e.g. Surface (or surface to node) pairs, there is nothing to stop both surfaces being everything in your model If you use (for droptest) /INTER/TYPE24 then you may specify a single surface for everything in your model as Surf1 (which will provide general self contact), you still need to specify what is 'in' that surface, but you can e.g. define it by box that encloses your whole model or by 'SUBSET' containing your whole model so you don't have to manually add all pars to a set. see: [Interface Type 24 in help](https://help.altair.com/hwsolvers/rad/topics/solvers/rad/inter_type24_starter_r.htm)

**Reply by li2qn** [reply]
Thanks for your help Paul. I was tried to add contact using keyword "/INTER/TYPE19/", and defined the external surfaces for solid elements with "/SURF/GRBRIC/FREE" as well surfaces from shell element with "/SURF/SEG". The simulation can run, but the result is definitely not correct. I suspect I was not using the contact card correctly, but I am not sure where. Could you please take a look if possible, I am still learning Radioss.  If look at my results, there is a big "spike" of element penetrated through the rigid wall and all solid elements just blown-up eventually. The foam material may not defined accurately, but even if I switched it to just elastic material, all solid element will have serious hourglasses. Appreciated any helps/comments 
[debug_contact.zip](https://github.com/OpenRadioss/OpenRadioss/files/11746459/debug_contact.zip)
![Spike](https://github.com/OpenRadioss/OpenRadioss/assets/48695377/96b8872c-52d3-47ec-84c3-6a068aaf98eb)


Regards
Lee

**Reply by PaulAltair** [reply]
Hi, attached are my iterations to run your model (there are a couple of changes you might want to modify once you fix some stuff)

4 runs each improving the model

1st modification: your node 1066 had bad formatting of the node locations (Y coordinate has a spurious extra decimal point), so in fact, it was already spiking out of the model before you even drop it, I deleted the extra point and mesh looks better

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/d0b63604-91f2-4c7a-a889-5e80ecf6e3f6)
 
![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/60400e5a-f8d8-4500-a3d9-36e5bb87e939)

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/156f00f7-f764-4c4d-9e24-8535df0663ae)

2nd modification: in your TYPE19 interface, you had 'gapmin' set as 1, but because your units are meters, this puts the parts into severe penetrations, I modified to 0.0001m (0.1mm), also, with type19, you cannot have the edge of the shells directly on the solid face with no gap, I offset the nodes in the bottom of the slot by 0.1mm to allow the contact to work

3rd modification: Your material LAW33 definition is not good, the yield curve is only specified in tension, and goes to zero so your foam behaves as though it is very soft

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/404c5059-9b8b-4383-aef1-962abf6e3bc9)
 it should be of the form shown below defined in tension and compression and above 0 throughout (it is only acting to cap the visco elastic behaviour from the modulus)

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/63b0e8c7-47c4-439a-883b-6a305ff8338c)

I switched the foam to use an elastic material with modulus 300000 and switched element formulation for the foam to ISOLID=24 to prevent hourglassing, this model runs ok!

4th modification: Your interface TYPE19 definition, while working, is not 'general contact' you have specified only that the foam and PCB should contact each other symmetrically (i.e. you have foam as 1 surf and PCB as another) so they would not contact themselves (in this loadcase they don't need to, but since you were asking about general contact I'm pointing this out), so in 4th model, I have switched to interface TYPE24 with a single surface definition, you can see, defining the surface is as easy as having 1 '/SURF/PART/EXT' listing your parts, and you reference it as SurfID1 in the contact

![image](https://github.com/OpenRadioss/OpenRadioss/assets/98418895/2c0f089f-d31e-46da-94d0-f1291b6866ae)

Hope this is helpful: The 4 model iterations and results are in the zip

[universal_contact_150623.zip](https://github.com/OpenRadioss/OpenRadioss/files/11756102/universal_contact_150623.zip)

**Reply by li2qn** [reply]
Thanks so much for your time and investigation. That's so helpful. Learned a lot here. Thanks again

**Reply by MarianBulla** [reply]
Yes, in addition to what Paul already mentioned, you can also buid a box in space and let the solver collect everything, which is inside to be treated as a possilble contct candidate.
You can also supstract parts from this box, afterwards.
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/24c88071-2408-410b-8917-ff785fa44153)
![image](https://github.com/OpenRadioss/OpenRadioss/assets/105412502/4dce8b5d-9d55-4b68-b7e6-649bb54797ee)

(here using the type7 interface, but it works with the newer type25 contact interface, too)

Hope it helps you ?

Thanks and best
Marian

**Reply by li2qn** [reply]
Thanks so much for your time and investigation. That's so helpful. Learned a lot here. Thanks again

**Reply by li2qn** [reply]
@PaulAltair @MarianBulla Thanks for all your help. It's so help. May I ask is there any material law that equivalent to Crushable foam in ABAQUS? In ABAQUS, I can use Crushable foam plasticity model and given foam hardening data. Thanks in advanced

**Reply by MarianBulla** [reply]
Yes, please have a look at LAW33.

Thanks and best regards,
Marian
